modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerWithSecurity.java

```diff
@@ -86,6 +87,10 @@ public AssetManagerWithSecurity(
     // Allow this if:
     //  - no previous snapshot exists
     //  - the user has write access to the previous snapshot
+    if (r.getSize() < 1) {
```

Comment:
If the user does not have write permissions we still nuke the old properties?

Reply:
A non-existent snapshot automatically means that you have write permissions. There are no permissions to check against after all which could forbid you from writing something. Look a few lines down and you will find:

https://github.com/opencast/opencast/blob/c93265c87de283ffcca825c1d5c68648a928f907/modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerWithSecurity.java#L94-L97